src/modules/dynamic-ui-wrapper/dynamic-ui-wrapper.ts

```diff
@@ -75,9 +80,10 @@ export class DynamicUiWrapper {
       return;
     }
 
-    this.currentControlType = newUserTask.data.preferredControl;
+    this.showConfirm = newUserTask.data.preferredControl !== undefined;
+    this.showForms = newUserTask.data.preferredControl === undefined;
```

Comment:
Same here. I suggest to test for 

```suggestion
    this.showForms = !this.showConfirm;
```

This clearly indicates, that `showForms` and `showConfirm` are exclusive AND if you add:

`this.showConfirm = newUserTask.data.preferredControl === "confirm";`

to the line above, it becomes very clear what is going on here.